Source/WebCore/layout/formattingContexts/inline/text/TextUtil.cpp

```diff
@@ -483,7 +483,48 @@ bool TextUtil::containsStrongDirectionalityText(StringView text)
     if (text.is8Bit())
         return false;
 
-    if (text.containsOnly<isNotBidiRTL>())
+    if (![&](auto span) ALWAYS_INLINE_LAMBDA {
+        using UnsignedType = std::make_unsigned_t<typename decltype(span)::value_type>;
```

Comment:
This is all very cool, but also a bit challenging to understand. I think these SIMD uses could really benefit from a block comment at the their head explaining what the following code is intending to do (I also find ascii art helps for understanding SIMD as well).

Reply:
Yeah, right now it exists on SIMDHelpers.h.

Reply:
I guess I meant something more along the lines of describing what the use of the SIMD operations is doing, rather than how operations work themselves. 

Take the scanText() function in the HTMLFastPathParser.  I was thinking something as simple as a comment saying something like:

"This function scans for the characters '<', '&', '\r' and '\0' in `m_parsingBuffer` using SIMD to perform the checks on strides of 16 characters at a time."

While not something that we would do in an ordinary code, I think the novelty of using SIMD makes a bit more explanation worthwhile.